### PR TITLE
v2.0.0 | Backwards incompatible changes

### PR DESCRIPTION
**Pull request by xeffyr**
## What is brought by new major release:

- Default distro plug-ins are immutable now.

  They are not intended to be editable by user anymore. For customization, users will need to make own copy. This change aims at simplifying updates and avoiding cases where something was not updated for unknown reason.

- Simplified distro plug-in structure.

  URLs are now retrieved from array instead of function returning values in specific format.

- Enforced integrity checking.

  Integrity checking is necessary to ensure that downloaded content is valid and not corrupted.

- No distribution versioning anymore.

  Due to numerous requests I have added versioning previously, but it appears bad design choice and has been a step from the original `proot-distro` purpose anyway. This feature is now disabled. Proot Distro won't serve customization requests anymore.

  If user need a custom distribution version, it should create own distro plug-in.

  Instead of versioned distributions only LTS, latest stable or rolling distribution variant will be provided.

- Added support for ArchLinux on i686 devices.

  Available as `archlinux` but really it is ArchLinux32 project.

- Added new distribution "OpenSUSE Tumbleweed".

  Available as `opensuse`.

- Own rootfs archives.

  PRoot Distro will now use own rootfs archive variants instead of the official.

  Script for generating tarballs is published.

- Rootfs customization is performed by distro plug-ins.

  Avoid editing /etc content during rootfs build time.

- Stabilizing design & feature set.

  It is not possible to make software suitable for everyone's needs while
  maintaining same quality. Since v2.x I will accept only feature requests
  that fit into PRoot Distro objectives and do not describe redundant
  functionality (i.e. that can be achieved through already existing).

  PRoot Distro has been intended to be simple both by design and usage.

***

Closes https://github.com/termux/proot-distro/issues/71 - OpenSUSE Tumbleweed is now included
Closes https://github.com/termux/proot-distro/issues/72 - ArchLinux32 (x86) is now included
Closes https://github.com/termux/proot-distro/issues/67 - pacman-key is now executed during rootfs build

### CODE REVIEW

File: distro-plugins/distro.sh.sample [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Default value is set by proot-distro script and is equal to the CPU
+# architecture of your device. You can set this to a custom value to
+# force use emulation mode (QEMU user).
+# Valid values are: aarch64, arm, i686, x86_64.
+#DISTRO_ARCH=aarch64
+
+# Name of distribution.
+DISTRO_NAME="Example"
+
```
Comment (Bills81) [inline distro-plugins/distro.sh.sample:9]: -  @SaicharanKandukuri

**Comment by SaicharanKandukuri** [comment]
@xeffyr there are some with armhf hirsute which I mentioned here #92 
https://github.com/termux/proot/issues/172

have you found any solution?

**Comment by xeffyr** [comment]
This is problem with `proot` specifically on your device model. It is possible to use Ubuntu 20.04 instead, but that is rather a problem avoidance than solution. Also no guarantee that someone else will not get problem with proot on it.

**Comment by SaicharanKandukuri** [comment]
>  on your device model
it's not my device anyway 
people who used my git repo to run hirsute in proot-distro reported me about this!
> no guarantee
almost every `armhf` user reported the same
> It is possible to use Ubuntu 20.04 instead
Good idea for now

**Comment by SaicharanKandukuri** [comment]
Looks like `armhf` proot can't run Ubuntu 21+ distros
I saw results of `Impish` dev branch tests results are same  as hirsute  
> fstat error

May be its because of new implementations in Ubuntu distros 
@xeffyr  What is the reason for this ( don't say device again )

**Comment by Grimler91** [comment]
@SaicharanKandukuri Please update the information in your proot issue instead of commenting here, since the issue is not directly related to proot-distro v2.0.0

**Comment by SaicharanKandukuri** [comment]
Ok
But issue is so stale

**Comment by ddos-pro** [comment]
Support for i686 is a lie, on Samsung galaxy tab 3 with Android 7, it does not install and says you have i686...
I installed Kali Linux, it installed even though it was called i386, and some other Linux systems were installed under i686

**Pull request merged**